### ISSUE TITLE
Bug: [no-unsafe-assignment] "Unsafe assignment of an error typed value." obscures underlying TS errors

### ISSUE THREAD

**Issue by bhollis** [typescript-eslint/typescript-eslint#9771]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.tsx&code=MYewdgzgLgBAZiEMC8MDeAoGMIgLYCmUAFgJZgDmAXDAOSggA2tGAvgNwYaiSwBGAQwBOKeIgB0YAYU4YgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZolp9kySgHMmAW0Qt0URNGgdokcGAC%2BITUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const foo = {
  something: 'cool'
};

const bar = foo.name;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unsafe-assignment": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I'd expect that I would see a TypeScript error about `name` not being a property of `foo`:

<img width="749" alt="A screenshot of VSCode's intellisense error tooltop saying that property 'name' does not exist on type '{ something: 'cool' }'" src="https://github.com/user-attachments/assets/266e9489-a785-4880-940c-a092f9abe0f3">

### Actual Result

I see only the rather unhelpful error from typescript-eslint that I have an "Unsafe assignment of an error typed value.":

<img width="823" alt="A screenshot of VSCode's intellisense error tooltop saying 'Unsafe assignment of an error typed value.'" src="https://github.com/user-attachments/assets/49677b76-8210-49fd-a707-74ab2abf8c4a">

The actual error is there if I happen to hover over just `.name`, but the underline is for the entire expression so I likely won't see the actual error.

### Additional Info

I do not want to disable `@typescript-eslint/no-unsafe-assignment` because it is a useful lint, but I find myself having to disable it as I work so I can see the real errors.

Futhermore, telling me that I'm assigning an "error typed value" is never useful to me, because there ought to always be an actual error highlighted somewhere that is producing the "error typed value".

**Comment by Josh-Cena** [comment]
We don't mirror TypeScript errors. This seems to be an IDE-specific thing and if you run `tsc` you get the real error. It has always behaved like that since the first day of these rules and I fail to see how we can make it better.

**Comment by JoshuaKGoldberg** [comment]
Two things come to mind:

* We could skip reporting in this case ... but then we run the risk of incorrect ESLint setups silently not catching problems they should
* This is why I always configure my ESLint warnings to a different squiggly color!
  * I've been meaning to talk to @bradzacher about this for _ages_ now 😄 

I'd suggest both should be FAQs somewhere. This comes up once in a while.

**Comment by Josh-Cena** [comment]
> This is why I always configure my ESLint warnings to a different squiggly color!

You can do that?!

Like you can specify an arbitrary color or do you just use the warning/info color?

**Comment by bradzacher** [comment]
Josh specifically downgrades them from error/warning to warning/error
I do not approve, personally 😅

> Futhermore, telling me that I'm assigning an "error typed value" is never useful to me, because there ought to always be an actual error highlighted somewhere that is producing the "error typed value".

That's not necessarily true! Because we aren't always in sync with TS in your IDE the error could be invisible to you.
We don't know what errors TS is emitting in your IDE vs what errors TS is emitting for us.
Though worth noting that we don't even know that latter set! From experience asking TS to calculate the errors can take tens of milliseconds - which adds up over a lint run - so we don't ask for it.

So we don't know what errors TS has emitted in your IDE and we don't know what errors TS has emitted for us. We just know that an `any` is an "error" any.

It's just a really difficult thing to work with, sadly.

**Comment by JoshuaKGoldberg** [comment]
You can set it manually or downgrade it manually 😄 https://github.com/microsoft/vscode-eslint/pull/1164

```jsonc
// .vscode/settings.json
{
  "eslint.rules.customizations": [{ "rule": "*", "severity": "warn" }]
}
```

**Comment by kirkwaiblinger** [comment]
I put the following variant in my user-level config

> ```jsonc
> {
>   "eslint.rules.customizations": [{ "rule": "*", "severity": "downgrade" }]
> }
> ```

Single greatest user experience improvement for linting ♥️ (shoutout to @JoshuaKGoldberg for implementing)

**Comment by bradzacher** [comment]
I personally use [this extension](https://marketplace.visualstudio.com/items?itemName=JoeBerria.statusbarerror) which has a feature to show error messages at the end of the line.
Which means that you can glance and tell what the error is without hovering. So the error level isn't a necessary marker to differentiate things.

**Comment by bhollis** [comment]
Thanks for the suggestions - they definitely help to distinguish between these errors. That said, I still think this is a confusing default, and it's something that changed somewhat recently (perhaps due to improved handling of error-any).

**Comment by bradzacher** [comment]
Note that the rule logic didn't change at all here - it's not reporting more or less than it used to! All we did was updated the rule so that it uses a different message if it detects an "error any".

So there's an interesting cognitive shift caused by that change quoting @Josh-Cena from our internal team chat:

> before the change people were just treating these "error-typed values" as sort of "ok ts-eslint doesn't special handle this case, it's just listening to TS, that's fine, I know its limitations" and now they are like "ts-eslint actually knows stuff, why is it still yelling at me when it can kindly step aside??"

**Comment by bradzacher** [comment]
It's also worth noting that we can't actually just ignore these "error anys" completely. This class of `any` can be created by purposely ignored errors. If we ignore it then it's possible for suppressed errors to leak `any` well beyond the suppression line.

```ts
// @ts-ignore
const x = a.b;
const y = x; // rule reports an error any assignment
```
[playground](https://typescript-eslint.io/play/#ts=5.5.2&fileType=.tsx&code=PTAEAEBcGcFoEsDmA7A9gJwKYCgDGrlpJQAPUAXlAEMA6AIwG48CjQBPC0poA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZolp9kySgHMmAW0Qt0URNGgdokcGAC%2BITUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

**Comment by JoshuaKGoldberg** [comment]
Closing out as it looks like this has stalled without a real next step. Someone yell at me if there's an action we should take.
